### PR TITLE
Hypr: new port in x11

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

New port

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 14.4
Xcode 15.3

macOS 10a190
Xcode 3.2

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by barracuda156** [comment]
@herbygillot This is good to merge, I guess.

File: x11/Hypr/Portfile [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+PortSystem          1.0
+PortGroup           cmake 1.1
+PortGroup           github 1.0
+PortGroup           legacysupport 1.1
+
+# _strlen, _strdup
+legacysupport.newest_darwin_requires_legacy 10
+
+github.setup        hyprwm Hypr 855ef848fe0aded503b4100e734e58a3b872a91f
```
Comment (reneeotten) [inline x11/Hypr/Portfile:11]: as usual, this should be using a tagged version. If you want to provide a tag that should be a -devel port.

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: x11/Hypr/Portfile (hunk 1 above)
Comment (barracuda156) [reply x11/Hypr/Portfile:11]: It does not build from the tag, see the issue. It uses a function which does not exist in MacOS.

P. S. I have a reason to make it from a commit when I do so with my ports.

Comment (barracuda156) [reply x11/Hypr/Portfile:11]: @reneeotten Could we please merge this to reduce the queue? (There is nothing Python here.)

Comment (barracuda156) [reply x11/Hypr/Portfile:11]: Upstream agreed to make a release, so I will update to that: https://github.com/hyprwm/Hypr/issues/127#issuecomment-2024280099

Comment (barracuda156) [reply x11/Hypr/Portfile]: @reneeotten I have switched to released version now.

**Comment by barracuda156** [comment]
@reneeotten Thank you!

**Pull request merged**